File: src/content/3.4/code/scala/snippet17.scala
def toWords: String => Writer[String, List[String]] =
  s => Writer(s.split("\\s+").toList, "toWords ")

def process: String => Writer[String, List[String]] = {
  import kleisliSyntax._
  // -Ypartial-unification should be on
  upCase >=> toWords
}

File: src/content/3.4/code/ocaml/snippet17.ml
let up_case s = Writer (String.uppercase_ascii s, "up_case ")
let to_words s = Writer (String.split_on_char ' ' s, "to_words ")

module Writer_Process
    (W : Monad with type 'a t = (string, 'a) writer) =
struct
  open W

  let process = up_case >=> to_words
end
